packages/client/src/index.ts

```diff
@@ -69,46 +69,52 @@ export default class LogtoClient {
     redirectUri: string,
     onRedirect: (url: string) => void = createDefaultOnRedirect()
   ) {
-    const { url, codeVerifier } = await getLoginUrlAndCodeVerifier({
+    const { url, codeVerifier, state } = await getLoginUrlAndCodeVerifier({
       baseUrl: this.oidcConfiguration.authorization_endpoint,
       clientId: this.clientId,
       scope: this.scope,
       redirectUri,
     });
-    this.sessionManager.set({ redirectUri, codeVerifier });
+    this.sessionManager.set({ redirectUri, codeVerifier, state });
     onRedirect(url);
   }
 
   public isLoginRedirect(url: string) {
     try {
-      const { code, error } = parseRedirectCallback(url);
+      const { code, state, error } = parseRedirectCallback(url);
+      if (!code || !state || error) {
+        return false;
+      }
 
-      if (error || !code) {
+      const session = this.sessionManager.get();
+      if (!session) {
         return false;
       }
-    } catch {
-      return false;
-    }
 
-    const session = this.sessionManager.get();
+      const { redirectUri, state: originalState } = session;
+      if (state !== originalState) {
```

Comment:
do this check only in `handleCallback` is enough, there we only need to check `state` exists

Reply:
Fixed.